compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/split/creators/JvmFunctionCallsCreatorsGen.java

```diff
@@ -0,0 +1,169 @@
+                    jvmPackageGen.lookupBIRFunctionWrapper(packageName + encodedMethodName);
+            if (bTypesCount % MAX_CALLS_PER_FUNCTION_CALL_METHOD == 0) {
+                mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, callMethod, FUNCTION_CALL, null, null);
+                mv.visitCode();
+                defaultCaseLabel = new Label();
+                int remainingCases = functions.size() - bTypesCount;
+                if (remainingCases > MAX_CALLS_PER_FUNCTION_CALL_METHOD) {
+                    remainingCases = MAX_CALLS_PER_FUNCTION_CALL_METHOD;
+                }
+                List<Label> labels = JvmCreateTypeGen.createLabelsForSwitch(mv, funcNameRegIndex, functions,
+                        bTypesCount, remainingCases, defaultCaseLabel, false);
+                targetLabels = JvmCreateTypeGen.createLabelsForEqualCheck(mv, funcNameRegIndex, functions,
+                        bTypesCount, remainingCases, labels, defaultCaseLabel, false);
+                i = 0;
+                callMethod = CALL_FUNCTION + ++methodCount;
+            }
+            Label targetLabel = targetLabels.get(i);
+            mv.visitLabel(targetLabel);
+
+            List<BType> paramTypes = func.type.paramTypes;
+            BType retType = func.type.retType;
+
+            // load strand
+            mv.visitVarInsn(ALOAD, 0);
+            int j = 0;
+            for (BType paramType : paramTypes) {
+                // load parameters
+                mv.visitVarInsn(ALOAD, 2);
+
+                // load j parameter
+                mv.visitLdcInsn((long) j);
+                mv.visitInsn(L2I);
+                mv.visitInsn(AALOAD);
+                jvmCastGen.addUnboxInsn(mv, paramType);
+                j += 1;
+            }
+            mv.visitMethodInsn(INVOKESTATIC, functionWrapper.fullQualifiedClassName, func.name.value,
+                    functionWrapper.jvmMethodDescription, false);
+            if (retType == null || retType.tag == TypeTags.NIL || retType.tag == TypeTags.NEVER) {
+                mv.visitInsn(ACONST_NULL);
+            } else {
+                jvmCastGen.addBoxInsn(mv, retType);
+            }
+            mv.visitInsn(ARETURN);
+            i += 1;
+            bTypesCount++;
+            if (bTypesCount % MAX_CALLS_PER_FUNCTION_CALL_METHOD == 0) {
+                if (bTypesCount == (functions.size())) {
+                    createDefaultCase(mv, defaultCaseLabel, funcNameRegIndex, "No such function: ");
+                } else {
+                    mv.visitLabel(defaultCaseLabel);
+                    mv.visitVarInsn(ALOAD, 0);
+                    mv.visitVarInsn(ALOAD, 1);
+                    mv.visitVarInsn(ALOAD, 2);
+                    mv.visitMethodInsn(INVOKESTATIC, functionCallsClass, CALL_FUNCTION + methodCount, FUNCTION_CALL,
+                            false);
+                    mv.visitInsn(ARETURN);
+                }
+                mv.visitMaxs(i + 10, i + 10);
```

Comment:
Is it better if we add this 10 as a constant like safe_margin_for_visitmaxs (suggesting a proper name)?

Reply:
Added a constant for every places.